STM32/I2C/Src/sht45.c

```diff
@@ -23,34 +31,15 @@ static HAL_StatusTypeDef sht4x_set_mode(sht4x_handle_t *dev, uint8_t command);
 ** PRIVATE FUNCTIONS
 ***************************************************************************************************/
 
-/*!
-** @brief Computes CRC of incoming data
-** @note CRC parameters can be found in section 4.4 of the datasheet 
-*/
-static uint8_t calculate_crc(const uint8_t *data, size_t length)
-{
-    uint8_t crc = 0xff;
-    for (size_t i = 0; i < length; i++) {
-        crc ^= data[i];
-        for (size_t j = 0; j < 8; j++) {
-            if ((crc & 0x80u) != 0) {
-                crc = (uint8_t)((uint8_t)(crc << 1u) ^ 0x31u);
-            } else {
-                crc <<= 1u;
-            }
-        }
-    }
-    return crc;
-}
-
 /*!
 ** @brief Compares computed and received CRCs
 ** @note For every transfer there are two CRCs.
 */
 static int checkCRC(uint8_t *buffer)
 {
     // Ensure data is valid
-    if (calculate_crc(buffer, 2) != buffer[2] || calculate_crc(buffer + 3, 2) != buffer[5]) {
+    initCrc8(CRC_INIT, CRC_POLY);
+    if (crc8Calculate(buffer, 2) != buffer[2] || crc8Calculate(buffer + 3, 2) != buffer[5]) {
         return 1;
```

Comment:
-1 would be a better return value if the CRC check fails but it would require modifications in the projects using this device so it's probably not worth it